rotkehlchen/chain/ethereum/modules/compound/v2/compound.py

```diff
@@ -103,21 +108,139 @@ def _get_apy(self, address: ChecksumEvmAddress, supply: bool) -> FVal | None:
+            for balance_type in tokens_and_types[token_address]:
+                decoded_utilization = token_contract.decode(
+                    result=call_output[idx],
+                    method_name='getUtilization',
+                )[0]
+                rates_calls[balance_type].append((
+                    token_address,
+                    token_contract.encode(
+                        method_name=f'get{"Supply" if balance_type == BalanceType.ASSET else "Borrow"}Rate',  # noqa: E501
+                        arguments=[decoded_utilization],  # passing decoded utilization
+                    ),
+                ))
+                rates_calls_arguments[balance_type].append(decoded_utilization)
+
+        try:
+            call_output = self.ethereum.multicall(  # get n supply and m borrow rates
+                calls=rates_calls[BalanceType.ASSET] + rates_calls[BalanceType.LIABILITY],  # n + m
+            )
+        except RemoteError as e:
+            log.error(f'Failed to query Compound v3 Supply and Borrow rates due to {e!s}')
+            return rates
+
+        for idx, (token_address, _) in enumerate(rates_calls[BalanceType.ASSET]):
+            rates[token_address][BalanceType.ASSET] = (FVal(token_contract.decode(
+                method_name='getSupplyRate',
+                arguments=[rates_calls_arguments[BalanceType.ASSET][idx]],
+                result=call_output[idx],  # first n are supply rates
+            )[0]) / ETH_MANTISSA) * YEAR_IN_SECONDS
+
+        for idx, (token_address, _) in enumerate(rates_calls[BalanceType.LIABILITY]):
+            rates[token_address][BalanceType.LIABILITY] = (FVal(token_contract.decode(
+                method_name='getBorrowRate',
+                arguments=[rates_calls_arguments[BalanceType.LIABILITY][idx]],
+                result=call_output[len(rates_calls[BalanceType.ASSET]) + idx],  # next m are borrow rates  # noqa: E501
+            )[0]) / ETH_MANTISSA) * YEAR_IN_SECONDS
+
+        return rates
+
+    def _get_v3_rewards(
+            self,
+            users: list[ChecksumEvmAddress],
+    ) -> dict['EvmToken', dict[ChecksumEvmAddress, FVal]]:
+        """Make multicalls to the compound v3 reward contracts to get the rewards of the users.
+        Returns them as a dict of token -> user_address -> rewards."""
+        rewards: dict['EvmToken', dict[ChecksumEvmAddress, FVal]] = defaultdict(lambda: defaultdict(FVal))  # noqa: E501
+        reward_contract = EvmContract(
+            address=ZERO_ADDRESS,  # not used here
+            abi=self.ethereum.contracts.abi('COMPOUND_V3_REWARDS'),
+            deployed_block=0,  # not used here
+        )
+        calls: list[tuple['ChecksumEvmAddress', str]] = []
+        token_arguments: list[EvmToken] = []
+        user_address_arguments: list['ChecksumEvmAddress'] = []
+        for token in (
+            EvmToken('eip155:1/erc20:0xc3d688B66703497DAA19211EEdff47f25384cdc3'),  # cUSDCv3
```

Comment:
why this specific 2 tokens? What is this?

Reply:
These are the only two compound v3 tokens on ethereum mainnet. Docs: https://docs.compound.finance/#networks